actions/recipe_action.go

```diff
@@ -58,19 +62,26 @@ func (recipe *RecipeAction) Verify(context *debos.DebosContext) error {
 
 	// Add Variables to template vars
 	for k, v := range recipe.Variables {
+		/* By default included recipe shared the same arhitecture with its
+		   parent. Overwrite the local context "architecture" in case this
+		   is set explicitly in Variables options */
+		if k == "architecture" {
+			recipe.context.Architecture = v
+		}
+
```

Comment:
This probably deserved splitting out in its own commit; With a better explanation what it's trying to do as i'm not entirely clear what the semantics of this are..

This seems to try and assert that the architecture is the same as the parent architecture unless it was explicitly set in the Variables for the recipe?